nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/TrashAction.java

```diff
@@ -120,6 +123,13 @@ public void setSystemProperty(CoreSession session, List<String> ids, Boolean val
                     DocumentRef ref = new IdRef((String) map.get(ECM_UUID));
                     try {
                         session.setDocumentSystemProp(ref, "isTrashed", value);
+                        String docName = (String) map.get(ECM_NAME);
+                        TrashService trashService = Framework.getService(TrashService.class);
+                        if (!value && trashService.isMangledName(docName)) {
```

Comment:
i am wondering in a case where `value` is `null` you will end with a NPE, you can replace it by using `BooleaUtils.isFalse`if the need is to treat the `null` an d the `false` as the same.

Reply:
Value should never be null. We cannot decide in case of null because it's too important to be decided in place of the user.
It's better to fail here than infering a wrong value.

Reply:
- _Value should never be null_  we are not sure about it, we can find ourself with a `null` value.

- _It's better to fail here than infering a wrong value._: yes and no, in a case of forward / backward compatibility (mainly the case of backward we will break the client code) if a client call this method with `null` before this modification the code works and it set a `null` value in the `isTrashed` but now it will fail. i don't know if it is the need or not, at least if we want to keep this new code perhaps adding more javadoc on the method ...